client/pokemon/pokemon.ctrl.js

```diff
@@ -69,11 +71,12 @@ module.exports = class Pokemon {
       `${this.data.formName ? '-' + this.data.formName : ''}`;
   }
   parseAllProps () {
+    this.uploadGen = this.data.gen;
     this.parseBoxViewProps();
 
+    this.tidFull = this.data.tidFull;
```

Comment:
For consistency, I think maybe we should do a migration at some point. It seems unexpected for clients to receive different properties based on when something was uploaded.

Alternatively we could update pkparse to make assignReadableNames add tidFull if it's not already present.